### ISSUE TITLE
Parsing JSON for Any is order sensitive

### ISSUE THREAD

**Issue by gurgenh** [google/protobuf#1824]
Parsing `Any` types from JSON relies on the `@type` being the first element. E.g., parsing a boolean value into `Any` works fine for this input:

``` json
{
  "@type": "type.googleapis.com/google.protobuf.BoolValue",
  "value": false
}
```

However, it fails to parse if `value` and `@type` are reordered:

``` json
{
  "value": false,
  "@type": "type.googleapis.com/google.protobuf.BoolValue"
}
```

**Comment by xfxyjwf** [comment]
Are you using the C++ implementation?. Right now you will need to put "@type" as the first entry to work around this problem (all protobuf JSON serializers should already be doing this).

**Comment by gurgenh** [comment]
Yes, the C++ implementation. Protobuf serializers generate the JSON in the "correct" order, but then if the payload goes through a different parser/serializer (in our case [JSON::PP](http://search.cpan.org/~makamaka/JSON-PP-2.27400/lib/JSON/PP.pm)) the order may change.

**Comment by xfxyjwf** [comment]
This should already be fixed in 3.2.0.
